### PR TITLE
feat: move reconnect banner to a new component

### PR DESCRIPTION
**Pull request by LuseBiswas**
# Fixes Issue

**My PR closes #691 **

# 👨‍💻 Changes proposed(What did you do ?)
I just made another component i.e. ReconnectBanner.jsx in Components 📁 of Client and And applied it in BuddyMatcher.jsx by pasing props.

# ✔️ Check List (Check all the applicable boxes)

- [X] My code follows the code style of this project.
- [X] This PR does not contain plagiarized content.
- [X] The title and description of the PR is clear and explains the approach.

##  Note to reviewers

# 📷 Screenshots

### CODE REVIEW

**Comment by LuseBiswas** [comment]
@Dun-sin I checked all the things and I found out that there is restirictions in a Max-Character Limit in One Line. So I break the Line into several parts to encounter that error.
And there are some components which are not in use but declared so I removed that too.
Hope this will make the build perfect. Rest I still not able to get you vercel link you are saying to me. 

I know its sound silly but can you please provide to me. Actually I am new in Open Source Contribution. SO thats why.

**Comment by Dun-sin** [comment]
@LuseBiswas pls stop creating new prs and just edit and fix your current PR, google it if you don't know how

secondly, you didn't fill the PR properly and the compliance error is still there(look at what i said in the last PR you just closed)

File: client/src/components/BuddyMatcher.jsx [hunk 1]
```diff
@@ -258,28 +260,12 @@ const BuddyMatcher = () => {
...
-				>
-					Try again
-				</button>
-				<Link to="/" className="underline text-white hover:text-white text-lg">
-					Return Home
-				</Link>
-			</div>
-		</div>
+		<ReconnectBanner handleReconnect={handleReconnect}/>
+		//Changes made here with moviing reconnector to another component.
```
Comment (Dun-sin) [inline client/src/components/BuddyMatcher.jsx:264]: no need for the comment

File: client/src/components/BuddyMatcher.jsx [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
-import { PiPlugsLight } from 'react-icons/pi';
+//import { PiPlugsLight } from 'react-icons/pi';
 import { connectWithId, socket } from 'src/lib/socketConnection';
 
 import Anonymous from 'components/Anonymous';
+import ReconnectBanner from 'components/ReconnectBanner';
 import { useAuth } from 'src/context/AuthContext';
 import { useChat } from 'src/context/ChatContext';
-import { useNavigate, Link } from 'react-router-dom';
+//import { useNavigate, Link } from 'react-router-dom';
```
Comment (Dun-sin) [inline client/src/components/BuddyMatcher.jsx:10]: no need to comment it, git shows you the history already

File: client/src/components/BuddyMatcher.jsx [hunk 3]
```diff
@@ -1,12 +1,13 @@
 import { useState, useEffect, useRef, useCallback } from 'react';
 import { ThreeDots } from 'react-loading-icons';
-import { PiPlugsLight } from 'react-icons/pi';
+//import { PiPlugsLight } from 'react-icons/pi';
```
Comment (Dun-sin) [inline client/src/components/BuddyMatcher.jsx:3]: same here no need to comment it out, just remove it if you aren't using it

File: client/src/components/ReconnectBanner.jsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+          to="/"
+          className="underline text-white hover:text-white text-lg"
+        >
+					Return Home
+				</Link>
+			</div>
+		</div>
+	);
+};
+// Define the expected prop types
```
Comment (Dun-sin) [inline client/src/components/ReconnectBanner.jsx:29]: no need for a comment here either

File: client/src/components/ReconnectBanner.jsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import PropTypes from 'prop-types'; // Import the PropTypes package
```
Comment (Dun-sin) [inline client/src/components/ReconnectBanner.jsx:2]: no need for the comment here

File: package-lock.json
Comment (Dun-sin) [inline package-lock.json:1]: remove this file and don't delete it.

```
Find the original version of the file, either on your fork if you forked from the main repository, or on the repository from which you cloned your local repository.
Copy the entire content of the file and paste it into your copy of the file.
Commit the new changes if necessary:
If you had not yet committed your unwanted changes, then you don’t need to do any more.
If you’d already committed your unwanted changes, create another commit now. The new commit wipes out your previous changes.
If you’d already pushed the changes up to a remote repository, push the new commit now too.
```

File: client/src/components/ReconnectBanner.jsx
Comment (Dun-sin) [inline client/src/components/ReconnectBanner.jsx:1]: pls run prettier locally, look at the package.json on the command to do that

**Review by Dun-sin — CHANGES_REQUESTED** [review]

**Comment by LuseBiswas** [comment]
@Dun-sin Thankyou for guiding me in detail. I had done all the changes as you said. Let me know if any other changes I need to do.

**Review by Dun-sin — CHANGES_REQUESTED** [review]
my mistake, pls remove all(don't delete) the files that you didn't work on

**Comment by Dun-sin** [comment]
@LuseBiswas also fill the PR template properly, you can edit it, check the 3 dots by the side of it

**Comment by LuseBiswas** [comment]
Hey @Dun-sin I just made all the changes:-
First:- I edited my PR Template by explaining my approcah and changes I made.
Second:- I remove (I dont delete it this time) all the files which I did not work on.

But there is an extra empty commit there named "[Merge resolved: Your commit message here]" it is done by me intentionally, actually I was facing problem to reset the last commit so I had to force the commit. Kindly look at that while merging it.

File: package-lock.json
Comment (Dun-sin) [inline package-lock.json:1]: still need to remove this

**Comment by LuseBiswas** [comment]
@Dun-sin if any changes I need to make then let me know. I'll make that. Because of your support I learnt so much new things.🙂

File: client/src/components/BuddyMatcher.jsx [hunk 6]
```diff
@@ -258,28 +255,10 @@ const BuddyMatcher = () => {
...
-					Return Home
-				</Link>
-			</div>
-		</div>
+		<ReconnectBanner handleReconnect={handleReconnect} />
 	) : (
-		<Anonymous 
-			onChatClosed={startNewSearch}  
-		/>
+		//Changes made here with moviing reconnector to another component.
```
Comment (Dun-sin) [inline client/src/components/BuddyMatcher.jsx:260]: remove this comment

File: client/src/components/ReconnectBanner.jsx
Comment (Dun-sin) [inline client/src/components/ReconnectBanner.jsx:1]: don't think you ran prettier for this file, do you have prettier installed on your VSCode(it should automatically do this on each save)

**Review by Dun-sin — CHANGES_REQUESTED** [review]

**Comment by Dun-sin** [comment]
> Hey @Dun-sin I just made all the changes:- First:- I edited my PR Template by explaining my approcah and changes I made. Second:- I remove (I dont delete it this time) all the files which I did not work on.
> 
> But there is an extra empty commit there named "[Merge resolved: Your commit message here]" it is done by me intentionally, actually I was facing problem to reset the last commit so I had to force the commit. Kindly look at that while merging it.

don't worry about the commit, that's normal, i will handle that when i merge this. you still need to provide the screenrecording in your PR. It can be a screenshot like in this PR -> #683 or a screenrecoding like this PR -> #680

**Comment by LuseBiswas** [comment]
Hey @Dun-sin its been a day I am trying to get that reconnect banner option on the web app. I tried to being offline and decreasing my internet speed through Network tab from inspection. But I am unable to get that.

It is not like this problem I am facing on my local website. Its even in your hosted website too. I think you should check it out.

Screenshot 2024-09-09 at 3 16 26 AM

**Comment by Dun-sin** [comment]
Thanks, will look into it

**Comment by Dun-sin** [comment]
@LuseBiswas

**Comment by LuseBiswas** [comment]
@Dun-sin I am so sorry...but actually I was little busy with my academics....I will make it ASAP.

**Pull request closed without merging**